src/atomate2/abinit/sets/response.py

```diff
@@ -0,0 +1,66 @@
+"""Module defining response function Abinit input set generators."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING, Callable
+
+from abipy.abio.factories import (
+    ddepert_from_gsinput,
+    ddkpert_from_gsinput,
+    dtepert_from_gsinput,
+)
+from abipy.abio.input_tags import DDE, DDK, DTE, SCF
+
+if TYPE_CHECKING:
+    from pymatgen.io.abinit import PseudoTable
+
+from atomate2.abinit.sets.core import NonSCFSetGenerator, StaticSetGenerator
+
+__all__ = [
+    "DdkSetGenerator",
+    "DdeSetGenerator",
+    "DteSetGenerator",
+]
+
+
+@dataclass
+class DdkSetGenerator(NonSCFSetGenerator):
+    """Class to generate Abinit DDK input sets."""
+
+    calc_type: str = "DDK"
+    factory: Callable = ddkpert_from_gsinput
+    restart_from_deps: tuple = (f"{DDK}:1WF",)
+    prev_outputs_deps: tuple = (f"{SCF}:WFK",)
+    nbands_factor: float = 1.0  # TODO: to decide if nbdbuf or not
+
+
+@dataclass
+class DdeSetGenerator(StaticSetGenerator):
+    """Class to generate Abinit DDE input sets."""
+
+    calc_type: str = "DDE"
+    factory: Callable = ddepert_from_gsinput
+    pseudos: str | list[str] | PseudoTable | None = None
+    restart_from_deps: tuple = (f"{DDE}:1WF|1DEN",)
+    prev_outputs_deps: tuple = (f"{SCF}:WFK", f"{DDK}:1WF")
+    factory_prev_inputs_kwargs: dict | None = field(
+        default_factory=lambda: {"gs_input": (SCF,)}
+    )
+    user_abinit_settings: dict = field(
+        default_factory=lambda: {"irdddk": 1, "ird1wf": 0}
```

Comment:
I suppose this should work fine and is easy. However, I think there might be other kind of calculations that need the 1WF from a DDK. Is that correct? In that case it would be nice if this could handled while resolving the dependencies. 
Could a solution be to check the number following the 1WF (1WFxx) and determine if it is the perturbation of a DDK based on that? ([see abinit docs](https://docs.abinit.org/guide/respfn/#1-description-of-perturbations)). 
Anyway, if this is bothersome, maybe add a TODO pointing to the fact that if more jobs will need to set irdddk=1 it would be preferable to find a common solution.

Reply:
I don't know if other calculations might need the same modification... I will leave a TODO.

Reply:
ok, let's leave it like this for the time being.